src/ophyd_async/core/signal.py

```diff
@@ -54,6 +54,7 @@ def __init__(
         super().__init__(name)
         self._timeout = timeout
         self._init_backend = self._backend = backend
+        super().__init__(name)
```

Comment:
I think I agree with this being at the bottom of the function, please could you remove the `super` call at the top of the function?

Reply:
I'm not sure I agree, is there a specific reason why the timeout and backend need to be set prior to the super call? To me this looks like an out-of-place change in the context of the rest of the PR.